d2/src/main/java/com/linkedin/d2/xds/XdsClientImpl.java

```diff
@@ -70,18 +71,22 @@ public class XdsClientImpl extends XdsClient
   private ScheduledFuture<?> _readyTimeoutFuture;
   private final long _readyTimeoutMillis;
 
+  private final XdsClientJmx _xdsClientJmx;
+
   @Deprecated
   public XdsClientImpl(Node node, ManagedChannel managedChannel, ScheduledExecutorService executorService)
   {
     this(node, managedChannel, executorService, DEFAULT_READY_TIMEOUT_MILLIS);
   }
 
   public XdsClientImpl(Node node, ManagedChannel managedChannel, ScheduledExecutorService executorService,
-      long readyTimeoutMillis) {
+      long readyTimeoutMillis)
+  {
     _readyTimeoutMillis = readyTimeoutMillis;
     _node = node;
     _managedChannel = managedChannel;
     _executorService = executorService;
+    _xdsClientJmx = new XdsClientJmx();
```

Comment:
Nit: As a general rule, the use of `new` should be avoided and injection through the constructor should be favored. There could be another constructor that takes the jmx bean instance as input, and perhaps the existing constructor could instantiate it for now and call the new constructor. This could also help with writing some UTs in case we ever do that. Not really required as of now though, feel free to ignore.

Reply:
makes sense. I'll keep for now though for simplicity.